src/app/messages-page/messages-inbox/messages-inbox.component.ts

```diff
@@ -51,6 +82,190 @@ export class MessagesInboxComponent implements OnInit, OnChanges {
+              this.newMessagesFromPage = res.OrderedContactsWithMessages.length;
+            }
+          }
+        },
+        (err) => {
+          console.error(this.backendApi.stringifyError(err));
+        }
+      ).finally(() => {
+      this.fetchingMoreMessages = false;
+      }
+    );
+  }
+
+  _setMessagesFilter(tabName: any) {
+    // Set the request parameters if it's a known tab.
+    // Custom is set in the filter menu component and saved in local storage.
+    if (tabName == "My Holders") {
+      this.globalVars.messagesRequestsHoldersOnly = true;
+      this.globalVars.messagesRequestsHoldingsOnly = false;
+      this.globalVars.messagesRequestsFollowersOnly = false;
+      this.globalVars.messagesRequestsFollowedOnly = false;
+      this.globalVars.messagesSortAlgorithm = "time";
+    } else if (tabName == "All") {
+      this.globalVars.messagesRequestsHoldersOnly = false;
+      this.globalVars.messagesRequestsHoldingsOnly = false;
+      this.globalVars.messagesRequestsFollowersOnly = false;
+      this.globalVars.messagesRequestsFollowedOnly = false;
+      this.globalVars.messagesSortAlgorithm = "time";
+    } else if (tabName == "Custom") {
+      this.globalVars.messagesRequestsHoldersOnly =
+        this.backendApi.GetStorage('customMessagesRequestsHoldersOnly');
+      this.globalVars.messagesRequestsHoldingsOnly =
+        this.backendApi.GetStorage('customMessagesRequestsHoldingsOnly');
+      this.globalVars.messagesRequestsFollowersOnly =
+        this.backendApi.GetStorage('customMessagesRequestsFollowersOnly');
+      this.globalVars.messagesRequestsFollowedOnly =
+        this.backendApi.GetStorage('customMessagesRequestsFollowedOnly');
+      this.globalVars.messagesSortAlgorithm = this.backendApi.GetStorage('customMessagesSortAlgorithm');
+    }
+  }
+
+  _clearMessages() {
+    this.newMessagesFromPage = null;
+
+    // If messageResponse isn't set, we return
+    if (!this.globalVars.messageResponse) {
+      return;
+    }
+
+    this.globalVars.messageResponse.OrderedContactsWithMessages = [];
+    this.globalVars.messageResponse.UnreadStateByContact = {};
+    this.globalVars.messageResponse.NumberOfUnreadThreads = 0;
+  }
+
+  _handleTabClick(tabName: any) {
+    // Clear the current messages
+    this._clearMessages();
+
+    // Set the messages filters based on the tab
+    this._setMessagesFilter(tabName);
+
+    // Make sure the tab is set in the url
+    this.activeTab = tabName;
+    this.router.navigate([], {
+      relativeTo: this.route,
+      queryParams: { messagesTab: MessagesInboxComponent.TABTOQUERY[tabName] },
+      queryParamsHandling: "merge",
+    });
+
+    // Fetch initial messages for the new tab
+    this.loadInitialMessages();
+  }
+
+  _toggleSettingsTray() {
+    this.globalVars.openSettingsTray = !this.globalVars.openSettingsTray;
+  }
+
+  _settingsTrayIsOpen() {
+    return this.globalVars.openSettingsTray;
+  }
```

Comment:
this open settings tray logic seems like it would be better served living in local storage.